tools/windows_buildenv.bat

```diff
@@ -95,5 +109,70 @@ EXIT /B 0
     SET /P BUILDENV_NAME=<%MIXXX_ROOT%\cmake\windows_build_environment_name
     SET BUILDENV_NAME=!BUILDENV_NAME:PLATFORM=%PLATFORM%!
     SET BUILDENV_NAME=!BUILDENV_NAME:CONFIGURATION=%CONFIGURATION%!
-    SET RETVAL=%BUILDENV_NAME%
     GOTO :EOF
+
+:GENERATE_CMakeSettings_JSON
+REM Generate CMakeSettings.json which is read by MS Visual Studio to determine the supported CMake build environments
+    SET CMakeSettings=%MIXXX_ROOT%\CMakeSettings.json
+    IF EXIST %CMakeSettings% (
+        ECHO ### CMakeSettings.json exist: Rename old file to CMakeSettings__YYYY-MM-DD_HH-MM-SS.json ###
+        FOR /f "delims=" %%a in ('wmic OS Get localdatetime ^| find "."') do set DateTime=%%a
+        REN %CMakeSettings% CMakeSettings__!DateTime:~0,4!-!DateTime:~4,2!-!DateTime:~6,2!_!DateTime:~8,2!-!DateTime:~10,2!-!DateTime:~12,2!.json
+    )
+    ECHO ### Create new CMakeSettings.json ###
+    >>%CMakeSettings% echo {
+    >>%CMakeSettings% echo   "configurations": [
+    SET configElementTermination=,
+    CALL :Configuration2CMakeSettings_JSON off       Release
+    CALL :Configuration2CMakeSettings_JSON legacy    Release
+    CALL :Configuration2CMakeSettings_JSON portable  Release
+    CALL :Configuration2CMakeSettings_JSON fastbuild RelWithDebInfo
+    SET configElementTermination=
+    CALL :Configuration2CMakeSettings_JSON native    Release
+    >>%CMakeSettings% echo   ]
+    >>%CMakeSettings% echo }
+    GOTO :EOF
+
+:Configuration2CMakeSettings_JSON <optimize> <configurationtype>
+    >>%CMakeSettings% echo     {
+    >>%CMakeSettings% echo       "buildRoot": "${projectDir}\\build_!PLATFORM!__%1",
```

Comment:
Since we have now removed the build folder from our source tree we can move all buildRoots to build. 
This matches more to the standard cmake approach. 
build\msvc_x64-RelWithDebInfo
or such.

Reply:
To put all buildRoots to a subdirectory "build" makes sense for me!
The same applies for the installRoots, should the Name of this subdirectory "dist" or "install"?
RelWithDebInfo in the buildRoot would hardcode the configurationType into the directory name, which would be wrong after setting the configurationType-Switch: 
![grafik](https://user-images.githubusercontent.com/64457745/102119902-1968e300-3e42-11eb-8590-d541bcf2cbbd.png)

Reply:
What is the installRoot in sense of cpack?

I think the wix packed is generated by cpack -G WIX 

This will install Mixxx to 

```build/msc_x64-RelWithDebInfi/_CPackBuild/Windows/WIX```

Is this the folder?

Reply:
> Is this the folder?
I don't have such a folder. installRoot points to a folder where the build generates a clean copy with all files needed for install. Currently I've set it like this:
buildRoot: mixxx/build_x64__fastbuild
installRoot: mixxx/dist_x64__native
AFAIK it's not common to have installRoot as subdirectory of buildRoot. It's usually a parallel path.

Reply:
Ok, than let's stick with the standard.

How does cpack integrate with visual studio? Is it possible to create the installer from this visual studio Gui? Or is it an extra command.

Visual studio calls the configure path from cmake. This generates also the cpack confg files.

Can you verify if this works with the json file? Does cpack uses the installRoot files of does it require a full rebuild of all cpp sources?

Reply:
I looked for this: CMake and CPack are shipped with VisualStudio and work as intended. But WiX is a third-party packagewhich is not contained in the buildenv-Download. Therefore CPack always fails with:
`CMake Error at C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.18/Modules/Internal/CPack/CPackWIX.cmake:12 (message):
  Could not find the WiX candle executable.`

Reply:
Does this work out of the Box using NSIS? Would you mind to try both?

If you use the cmake Visual Studio Generator you get a PACKAGE target. Like described here: 
https://dominoc925.blogspot.com/2016/09/create-windows-installer-using-cmake.html
Do we have similar with the json file?

Reply:
I investigated this:
1.) WiX needs to be downloaded manually, because it's missing in the Mixxx buildenv download.
2.) Visual Studio ships cpack.exe, but there is no IDE integration for CPack yet
3.) cpack -G WIX works, if called from inside the Visual Studio "Developer Command Prompt"

Reply:
Cool thank you.
Does cpack do a full rebuild?

Reply:
No, it just creates the .msi file in mixxx\build_x64__fastbuild\_CPack_Packages\win64\WIX

Reply:
Good. So everything should be fine.

Reply:
I changed buildroot and installroot as sugested
I added the manual steps, to create an .msi installer using WiX